src/scripts/main.js

```diff
@@ -1,3 +1,495 @@
+
+  setTimeout(() => {
+    errorPopUp.style.display = 'none';
+  }, 2000);
+}
+
+function warning() {
+  const block = notification();
+
+  block.classList.add('warning');
+  block.children[0].textContent = 'Oops... The data is invalid';
+  block.children[1].textContent = `Please, check the data once again :(`;
+
+  return block;
+}
+
+function showWarning() {
+  const warningPopUp = warning();
+
+  setTimeout(() => {
+    warningPopUp.style.display = 'none';
+  }, 2000);
+}
+
+function convertToCurrency(number) {
+  const formatter = new Intl.NumberFormat('en-US', {
+    style: 'currency',
+    currency: 'USD',
+    maximumFractionDigits: 0,
+  });
+
+  return formatter.format(number);
+}
+
+function setNewValue(input, oldValue) {
+  let newCellValue = input.value;
+  const cell = input.parentNode;
+  let incorrectData = false;
+
+  if (checkLength(newCellValue)) {
+    incorrectData = true;
+  }
+
+  const cellIndex = [...cell.parentNode.children].indexOf(cell);
+
+  switch (cellIndex) {
+    case 0:
+      if (checkName(newCellValue)) {
+        incorrectData = true;
+      }
+      break;
+
+    case 3:
+      if (checkAge(newCellValue)) {
+        incorrectData = true;
+      }
+      break;
+
+    case 4:
+      if (checkSalary(newCellValue)) {
+        incorrectData = true;
+      }
+      break;
+  }
+
+  if (incorrectData) {
+    incorrectData = true;
+
+    showError();
+  }
+
+  if (oldValue.indexOf('$') !== -1) {
+    newCellValue = convertToCurrency(+newCellValue);
+  }
+
+  if ((newCellValue.length > 0) && (newCellValue !== '$0')
+  && (incorrectData === false)) {
+    cell.textContent = newCellValue;
+  } else {
+    cell.textContent = oldValue;
+  }
+
+  input.value = '';
+  input.remove();
+  countActiveCells = 0;
+}
+
+for (let i = 0; i < titles.length; i++) {
+  titles[i].addEventListener('click', e => {
+    sortTable(i);
+  });
+}
+
+tbody.addEventListener('click', e => {
+  const row = e.target.parentNode;
+  const activeElements = [...document.querySelectorAll('.active')];
+
+  if (!activeElements.length) {
+    row.classList.add('active');
+  } else if (activeElements.length === 1) {
+    row.classList.add('active');
+    activeElements[0].classList.remove('active');
+  }
+});
+
+const fieldsNames = ['Name', 'Position', 'Age', 'Salary'];
+const selectionItems = [
+  `Tokyo`, `Singapore`, `London`, `New York`, `Edinburgh`, `San Francisco`,
+];
+
+const form = createForm(body, 'new-employee-form');
+
+createInputs(form, fieldsNames);
+createSelection(form, 'Office', selectionItems);
+
+createSubmitButton(form);
+
+let userInputArr = [];
+
+form.addEventListener('submit', e => {
+  e.preventDefault();
+
+  let incorrectData = false;
+
+  for (let i = 0; i < form.children.length - 1; i++) {
+    const label = form.children[i];
+    let inputValue = label.children[0].value;
+
+    inputValue = inputValue.trim();
+
+    if (checkLength(inputValue)) {
+      showWarning();
+      incorrectData = true;
+    }
+
+    switch (i) {
```

Comment:
It would be better to get input's name attribute and switch based on that, it's much easier to understand code like
```
switch (inputName) {
  case "name":
  ...
}
```
than when everything is index based
Please fix in other cases too